File: test/fixtures/output/ocaml/cohttp/full.ml
open Cohttp_lwt_unix
open Lwt

let uri = Uri.of_string "http://mockbin.com/har?foo=bar&foo=baz&baz=abc&key=value" in
let headers = Header.init ()
  |> fun h -> Header.add h "cookie" "foo=bar; bar=baz"
  |> fun h -> Header.add h "accept" "application/json"
  |> fun h -> Header.add h "content-type" "application/x-www-form-urlencoded"
in
let body = "foo=bar" in

Client.call ~headers ~body (Code.method_of_string "POST") uri
>>= fun (res, body_stream) ->
  (* Do stuff with the result *)

File: src/targets/javascript/fetch/fixtures/full.js
const options = {
  method: 'POST',
  headers: {
    cookie: 'foo=bar; bar=baz',
    accept: 'application/json',
    'content-type': 'application/x-www-form-urlencoded'
  },
  body: new URLSearchParams({foo: 'bar'})
};

fetch('https://httpbin.org/anything?foo=bar&foo=baz&baz=abc&key=value', options)
  .then(res => res.json())
  .then(res => console.log(res))
  .catch(err => console.error(err));